os/arch/Kconfig

```diff
@@ -107,13 +107,27 @@ config ARCH_USE_MMU
 config ARCH_USE_MPU
 	bool "Enable MPU"
 	default n
+	select MPU_RESERVED
```

Comment:
When A is mandatory for B, then we use "select" syntax. But I don't think this case is fit.
How about removing this select and adding "depends on ARCH_USE_MPU" in MPU_RESERVED?
Because MPU_RESERVED could be unused with ARCH_USE_MPU, but ARCH_USE_MPU must be enabled to use MPU_RESERVED.

Reply:
Yes, I agree that would be more apt. I will modify